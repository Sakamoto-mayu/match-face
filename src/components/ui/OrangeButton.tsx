"use client";
import { ComponentProps } from "react";

type ButtonProps = {
  label: string;
} & ComponentProps<"button">;

const OrangeButton = ({ label, ...props }: OrangeBtnProps) => {
  // console.log(label);
  const style =
    "bg-amber-500 drop-shadow-lg hover:saturate-150 active:drop-shadow-none active:shadow-inner active:mt-0.5 text-white rounded-xl w-40 h-8 text-lg";
  return (
    <div>
      <button className={style} {...props}>
        {label}
      </button>
    </div>
  );
};

export default OrangeButton;
