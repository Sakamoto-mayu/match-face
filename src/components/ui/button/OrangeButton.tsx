'use client';
import { ComponentProps } from 'react';
import { twMerge } from 'tailwind-merge';

type OrangeBtnProps = {
  label: string;
} & ComponentProps<'button'>;

const OrangeButton = ({ label, ...props }: OrangeBtnProps) => {
  // console.log(label);
  const style = twMerge(
    'bg-orange drop-shadow-lg hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange text-white rounded-xl w-40 h-8 text-lg',
    props.className
  );
  return (
    <div>
      <button {...props} className={style}>
        {label}
      </button>
    </div>
  );
};

export default OrangeButton;
