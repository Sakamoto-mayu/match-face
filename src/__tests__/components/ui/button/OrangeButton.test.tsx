import OrangeButton from '@/components/ui/button/OrangeButton';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

test('スナップショットテスト', () => {
  const { container } = render(<OrangeButton label="OrangeButton" />);
  expect(container).toMatchSnapshot();
});

test('labelに与えた値がbuttonの値となる', () => {
  render(<OrangeButton label="OrangeButton" />);
  const button = screen.getByRole('button', { name: 'OrangeButton' });
  expect(button).toBeInTheDocument();
  // const label = screen.getByText('OrangeButton');
  // expect(label).toBeInTheDocument();
  screen.debug();
});
