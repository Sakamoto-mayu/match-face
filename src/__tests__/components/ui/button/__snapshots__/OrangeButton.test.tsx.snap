// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショットテスト 1`] = `
<div>
  <div>
    <button
      class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-lg"
    >
      OrangeButton
    </button>
  </div>
</div>
`;
